OpenRA.Game/Graphics/TerrainSpriteLayer.cs

```diff
@@ -227,6 +237,40 @@ public void Dispose()
 				worldRenderer.TerrainLighting.CellChanged -= UpdateTint;
 
 			vertexBuffer.Dispose();
+
+			lock (IndexBuffers)
+				indexBuffer.Dispose();
+		}
+	}
+
+	public sealed class IndexBufferRc : IDisposable
+	{
+		public IIndexBuffer Buffer;
+		int count;
```

Comment:
count *must* start at 1

Reply:
so that when there's only 1 reference it would have a value of 2?

Reply:
Ah the AddRef will handle it. Yes ignore me.